src/sql/workbench/services/query/common/queryRunner.ts

```diff
@@ -513,6 +517,25 @@ export default class QueryRunner extends Disposable {
 		};
 		this._onVisualize.fire(result);
 	}
+
+	private logPerfMarks(): void {
+		let marks = perf.getMarks().filter(mark => mark.name.startsWith(`sql/query/${this.uri}`));
+		// One query editor can have one set of marks for each query gets executed,
+		// we only log the latest set of marks for the latest query.
+		let reverseIdx = marks.slice().reverse().findIndex(m => m.name.startsWith(`sql/query/${this.uri}/runQuery`));
+		marks = marks.slice(-(reverseIdx + 1));
+		let head = `Name\tTimestamp\tDelta\tTotal\n`;
+		let traceMsg = '';
+		let lastStartTime = -1;
+		let total = 0;
+		for (const { name, startTime } of marks) {
+			let delta = lastStartTime !== -1 ? startTime - lastStartTime : 0;
+			total += delta;
+			traceMsg += `${name}\t${startTime}\t${delta}\t${total}\n`;
+			lastStartTime = startTime;
+		}
+		this.logService.debug(`Traces for sql/query/${this.uri}:\n${head}${traceMsg}`);
```

Comment:
should we be logging the timing details to telemetry?  or is that something we plan to look at later?

Reply:
I can add that in a later PR.